app/workers/extract_worker.rb

```diff
@@ -11,7 +11,10 @@ def perform(classification_id)
     return unless classification
 
     workflow = classification.workflow
-    return if workflow.paused?
+
+    # if reducible is only paused, continue processing everything but extracts
+    # if reducible is halted, do not process anything
```

Comment:
This doesn't test if they reducible is `halted?` at all, i assume based on the comments that it should...right?

Reply:
since the states "active" "paused" and "halted" are mutually exclusive, testing for `active?` implicitly tests `not paused? && not halted?`

when deciding whether to run extractors, we need to handle both cases the same way, by doing nothing; when deciding whether to reduce or to run rules, they need to be handled in different ways

Reply:
Ah I see -  I think i'd prefer the explict form of this logic rather than interpolating the enum state but i'll leave that up to you.

>  if reducible is only paused, continue processing everything but extracts
if reducible is halted, do not process anything

FWIW i think the comment could be clarified as this is an extract worker. Maybe reflect the comments you left on this PR instead of what is currently set.

Reply:
that's fine, i just copy and pasted the comments for clarity but i can add more